### PR TITLE
feat: add VERS gem versioning scheme support

### LINKED ISSUE #52: feat: add VERS gem versioning scheme support
## Problem Statement

The VERS specification supports RubyGems versioning scheme, but go-univers currently only implements Gem ecosystem support without VERS integration. Users cannot use VERS range notation with Gem versions like `vers:gem/>=1.2.0|<2.0.0`.

## Research

From the [VERS specification](https://github.com/package-url/purl-spec/blob/main/VERSION-RANGE-SPEC.rst), Gem (RubyGems) is "Similar to node-semver for its syntax" but "Does not use semver versions". The [univers Python implementation](https://github.com/aboutcode-org/univers) includes Gem support in its VERS implementation.

RubyGems uses its own versioning with pessimistic constraints (`~>`) and standard comparison operators.

## Current State

- ✅ Gem ecosystem exists at `pkg/ecosystem/gem/`
- ✅ Gem version parsing and comparison implemented
- ❌ No VERS integration for Gem versioning scheme
- ❌ Cannot parse/evaluate ranges like `vers:gem/>=1.2.0|<2.0.0`

## Implementation Strategy

Following the established pattern used for npm, pypi, go, and maven VERS schemes:

1. **Create `pkg/spec/vers/gem.go`** with:
   - `gemContains(constraints []string, version string) (bool, error)` function
   - `intervalToGemRanges(interval interval) []string` function

2. **Update `pkg/spec/vers/vers.go`**:
   - Add `"gem": gemContains` to `schemeToContains` map
   - Add `case "gem": rangeStrs = intervalToGemRanges(interval)` to `toRanges` function
   - Update supported ecosystems documentation

3. **Add comprehensive tests** in `pkg/spec/vers/gem_test.go`

## Success Criteria

- [ ] Support VERS Gem ranges: `vers:gem/>=1.2.0`
- [ ] Handle prerelease versions correctly: `vers:gem/>=1.0.0.pre1|<2.0.0`
- [ ] Support all VERS operators: `>=`, `<=`, `>`, `<`, `=`, `!=`
- [ ] Pass comprehensive test suite covering edge cases
- [ ] CLI integration: `univers vers contains "vers:gem/>=1.2.0" "1.5.0"`
- [ ] Consistent behavior with existing Gem ecosystem implementation

## Resources

- [VERS Specification](https://github.com/package-url/purl-spec/blob/main/VERSION-RANGE-SPEC.rst)
- [Univers Python Implementation](https://github.com/aboutcode-org/univers)
- [RubyGems Versioning Guide](https://guides.rubygems.org/patterns/#semantic-versioning)
- Existing implementations: [npm VERS](https://github.com/alowayed/go-univers/issues/35), [pypi VERS](https://github.com/alowayed/go-univers/issues/36), [go VERS](https://github.com/alowayed/go-univers/issues/37)
- Pattern reference: `pkg/spec/vers/npm.go`

### PR DESCRIPTION
**Pull request by alowayed**
## Summary

Implements VERS (Version Range Specification) support for the RubyGems ecosystem, enabling users to use VERS range notation with Gem versions like `vers:gem/>=1.2.0|<2.0.0`.

- **Add `gemContains()` function** for RubyGems VERS constraint checking
- **Add `intervalToGemRanges()` function** to convert intervals to gem range syntax  
- **Update vers.go** to include gem in `schemeToContains` map and `toRanges` switch
- **Comprehensive test suite** covering all VERS operators and edge cases
- **CLI integration** with command `univers vers contains "vers:gem/>=1.2.0" "1.5.0"`

## Implementation Details

Following the established pattern used for npm, pypi, go, maven, and alpine VERS schemes:
- Created `pkg/spec/vers/gem.go` with gem-specific VERS functions
- Updated `pkg/spec/vers/vers.go` to register gem support
- Added comprehensive tests in `pkg/spec/vers/gem_test.go`

Key implementation insight: RubyGems uses comma-separated constraints (not space-separated), so `intervalToGemRanges` generates ranges like `>=1.0.0,<=2.0.0`.

## Test Plan

- [x] All existing tests pass (no regressions)
- [x] New comprehensive test suite with 25 test cases
- [x] Code quality checks pass (golangci-lint)
- [x] CLI functionality verified

### Success Criteria Verification

- [x] **Support VERS Gem ranges**: `vers:gem/>=1.2.0` → ✅
- [x] **Handle prerelease versions**: `vers:gem/>=1.0.0.pre1|<2.0.0` → ✅  
- [x] **Support all VERS operators**: `>=`, `<=`, `>`, `<`, `=`, `!=` → ✅
- [x] **Pass comprehensive test suite** covering edge cases → ✅
- [x] **CLI integration**: `univers vers contains "vers:gem/>=1.2.0" "1.5.0"` → ✅
- [x] **Consistent behavior** with existing Gem ecosystem implementation → ✅

## Examples

```bash
# Basic range
univers vers contains "vers:gem/>=1.2.0|<=2.0.0" "1.5.0"  # → true

# Prerelease support  
univers vers contains "vers:gem/>=1.0.0.pre1|<2.0.0" "1.5.0"  # → true

# Exclusion operator
univers vers contains "vers:gem/>=1.0.0|!=1.5.0" "1.5.0"  # → false

# All operators supported
univers vers contains "vers:gem/>1.0.0|<2.0.0" "1.5.0"  # → true
```

Fixes #52

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: pkg/spec/vers/gem_test.go [hunk 1]
```diff
@@ -0,0 +1,199 @@
+package vers
+
+import (
+	"testing"
+)
+
+// TestContains_Gem tests VERS functionality specifically for the RubyGems ecosystem
+func TestContains_Gem(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (alowayed) [reply pkg/spec/vers/gem_test.go:8]: This goes against the established convention of only testing the public method.
